File: golden_ticket_enterprise/lib/entities/sub_tag.dart

import 'package:golden_ticket_enterprise/entities/main_tag.dart';

class SubTag {
  final int subTagID;
  final String subTagName;

  SubTag({ required this.subTagID, required this.subTagName});

  factory SubTag.fromJson(Map<String, dynamic> json) {
    dynamic userData = json;
    return SubTag(
        subTagID: userData['subTagID'],
        subTagName: userData['subTagName'] ?? ""
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'subTagID': subTagID,
      'subTagName': subTagName
    };
  }

}

File: GoldenTicket/GoldenTicket/Entities/SubTag.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using GoldenTicket.Entities;

namespace GoldenTicket.Entities
{
    [Table("tblSubTags")]
    public class SubTag
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int TagID { get; set; }
        [Required]
        public string? TagName { get; set; }
        [Required]
        public int MainTagID { get; set; }
        [ForeignKey("MainTagID")]
        public MainTag? MainTag { get; set; } = null;
    }
    public class SubTagDTO
    {
        public int SubTagID { get; set; }
        public string? SubTagName { get; set; }
        public SubTagDTO(SubTag subTag){
            this.SubTagID = subTag.TagID;
            this.SubTagName = subTag.TagName;
        }
    }

}
